dvc/scm/git/__init__.py

```diff
@@ -324,15 +327,30 @@ def get_tree(self, rev):
         return GitTree(self.repo, self.resolve_rev(rev))
 
     def get_rev(self):
-        return self.repo.git.rev_parse("HEAD")
+        return self.repo.rev_parse("HEAD").hexsha
 
     def resolve_rev(self, rev):
-        from git.exc import GitCommandError
-
+        from git.exc import BadName, GitCommandError
+        from contextlib import suppress
+
+        names = [rev] + ([] if Git.is_sha(rev) else ["origin/" + rev])
+        for name in names:
+            with suppress(BadName, GitCommandError):
+                try:
+                    # Try python implementation of rev-parse first, it's faster
+                    return self.repo.rev_parse(name).hexsha
+                except NotImplementedError:
```

Comment:
Hm, when does it happen though?

Reply:
For text search, see `test_diff.test_directories`. There are also dates in `@{...}` format.